### PR TITLE
Remove all the compatibility code for Python 2.x

### PR DESCRIPTION
**Pull request by Valinor**
See #8566

* [X] Title is descriptive/clear for inclusion in release notes.
* [x] Applied a `Component: xxx` label.
* [ ] Applied the `api-deprecation` or `api-break` label.
* [ ] Applied the `release-highlight` label to be highlighted in release notes.
* [x] Added to the milestone version it was merged into.
* [ ] **Unittests** are included in PR.
* [ ] Properly documented, including `versionadded`, `versionchanged` as needed.

### CODE REVIEW

File: kivy/tests/test_fbo_py2py3.py
Comment (misl6) [inline kivy/tests/test_fbo_py2py3.py:1]: Maybe we can rename the whole test file and rename `FBOPy2Py3TestCase` (if it makes sense to have this test) ?

File: kivy/uix/rst.py [hunk 1]
```diff
@@ -605,8 +604,6 @@ def _load_from_text(self, *largs):
             # parse the source
             document = utils.new_document('Document', self._settings)
             text = self.text
-            if PY2 and type(text) is str:
-                text = text.decode('utf-8')
             self._parser.parse(text, document)
```
Comment (misl6) [inline kivy/uix/rst.py:607]: Maybe:
```
self._parser.parse(self.text, document)
```

**Review by misl6 — CHANGES_REQUESTED** [review]
Hi @Valinor !

I've left some minor suggestions.

Nice work and cleanup 🧹 BTW!

**Comment by Valinor** [comment]
> Hi @Valinor !
> 
> I've left some minor suggestions.
> 
> Nice work and cleanup 🧹 BTW!

Thanks for the review. Shame on me to let pass all of these :').

I will try to understand why some tests fails on the ci, and not for me, and take care of your suggestions.

**Comment by misl6** [comment]
> I will try to understand why some tests fails on the ci

When you push changes, make sure to rebase on top of latest master, as the CI errors you seen might be related to https://github.com/kivy/kivy/pull/8618 (so, not your fault)

**Comment by misl6** [comment]
@Valinor FYI:

We also may want to do some cleanup here:
https://github.com/kivy/kivy/blob/dc7fd3d8ae28029ebecfbb923af37caa4a5245c9/kivy/uix/screenmanager.py#L1090

**Comment by misl6** [comment]
Hi @Valinor,

This is a friendly ping, are you still active on this PR?

**Comment by misl6** [comment]
@Valinor FYI: I've performed some changes to your PR due to inactivity, thank you for doing most of the work BTW!

**Comment by Valinor** [comment]
> @Valinor FYI: I've performed some changes to your PR due to inactivity, thank you for doing most of the work BTW!

Thanks, and sorry for my inactivity. I will try to be more productive next time. Great change, thanks for your work.

**Comment by Samael-TLB** [comment]
@misl6 
A question on your pr [Avoid trickery for next iterator](https://github.com/kivy/kivy/pull/8610/commits/b77587c9a1999fad4f90d76fdbda2911044d1595)

Why have we removed the api function to cleanly get the next screen from the screenmanager? As per current implementation, people will need to do next(sm) and not sm.next(), which feels a bit unintuitive in lieu of class based apis. Am I missing anything, can you help me understand the reason?

**Pull request merged**